src/components/Button.jsx

```diff
@@ -0,0 +1,44 @@
+import  React, {useState, useEffect} from 'react'
+import PropTypes from 'prop-types'
+import '../App.css'
+
+const  Button = (props) => {
+    const [wide, setWide] = useState('25%');
+    useEffect(()=>{
+        if(props.wide) setWide('50%');
+    });
```

Comment:
No need to use state for wide here as it doesn't change during the lifecycle of the component. The style could have been just:
```
width: props.wide ? '50%' : '25%',
```

Reply:
Great point thanks for the tip :100: